src/App.js

```diff
@@ -29,32 +66,260 @@ class App extends React.Component {
+                value={this.state.fileInputValue}
+                className="input"
+                onChange={(e) => {
+                  this.setState({
+                    fileInputFile: e.target.files[0],
+                    fileInputValue: e.target.value,
+                  });
+                }}
+              />
+              <br />
+            </div>
+            <button
+              type="submit"
+              disabled={!this.state.textInputValue}
+              className="submit"
+            >
+              {this.state.textInputValue ? (
+                <FontAwesomeIcon icon={faPaperPlane} bounce className="i" />
+              ) : (
+                <div className="input-group">Start Typing</div>
+              )}
+            </button>
+          </form>
+
+          <div>
+            {this.state.isLoggedIn ? (
+              <NewsFeed
+                messages={this.state.messages}
+                deleteMessage={this.deleteMessage}
+                likePost={this.likePost}
+              />
+            ) : (
+              <div></div>
+            )}
+          </div>
         </header>
       </div>
     );
   }
 }
 
 export default App;
+
+/** componentDidMount -> setting up a listener for new messages being added to the Firebase collection, keeping the React component's state in sync with the changes to the state's messages array
+ 
+ *   componentDidMount() {
+    const messagesRef = ref(database, DB_MESSAGES_KEY); // initializing a Firebase reference to the 'messages' collection in the Firebase database
+   
+    onChildAdded(messagesRef, (data) => {
+      // onChildAdded listens to any new child added to the 'messages' collection. 
+      
+      this.setState((state) => ({
+        // When a new child is detected, it updates the component's state by appending the new messages to the 'messages' array
+        
+        messages: [...state.messages, { key: data.key, val: data.val() }],
+      }));
+    });
+  }
+
+  - lifecycle method in React and is called after the component has been rendered in the DOM, making it a suitable place to perform tasks like fetching data, subscriptions and other operations once the component is mounted
+
+ */
+
+/** writeData -> handling submission of new messages to Firebase
+ *   writeData = (event) => {
+    event.preventDefault(); // prevents default form submission behaviour
+
+    const messageListRef = ref(database, DB_MESSAGES_KEY); // creates new reference to the 'messages' collection in Firebase 
+
+    const newMessageRef = push(messageListRef); // generates a new child node using push()
+
+    set(newMessageRef, {
+      // data (message and timestamp) is set under this new child node, newMessageRef, using set()
+
+      message: this.state.textInputValue,
+
+      date: new Date().toLocaleString(),
+    });
+
+    this.setState({ textInputValue: "" }); // clears textInputValue in the component's state to make the input field empty for the next message
+  };
+ 
```

Comment:
Don't keep such comment blocks. As mentioned earlier, keep a notebook for explanations :)